contrib/vdsm-dev

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env bash
+# Author: Albert Esteve <aesteve@redhat.com>
+
+venv_path=$HOME/.venv/vdsm
+venv_activate_path=${venv_path}/bin/activate
+user_storage_path=/var/tmp/vdsm-storage/
+
+print_help()
+{
+    echo "oVirt/vdsm command line helper."
+    echo "Should be run from the repository root folder."
+    echo "Usage: $0 CMD [options]"
+    echo
+    echo "Available commands:"
+    echo "help:          Print this message and exits."
+    echo "build:         Build vdsm and make RPMs."
+    echo "build-install: Build vdsm, make RPMs, and update local packages."
+    echo "test:          Run any test from specified environment using tox."
+    echo "test-storage:  Run one/all storage tests using tox."
+    echo "show-coverage: Show coverage results generated after a test run."
+    echo "               Receives an environment with '-e env'. [default=storage]"
+}
+
+venv_create()
+{
+    echo "Creating venv at $venv_path"
+    python3 -m venv ${venv_path}
+    source "${venv_activate_path}"
+    pip install --upgrade pip
+    pip install -r docker/requirements.txt
+    deactivate
+}
+
+make_storage()
+{
+    if [ ! -d $user_storage_path ]; then
+        echo "Make storage at $user_storage_path"
+        make storage
+    fi
+}
+
+test_path()
+{
+    path=$1
+    storage_prefix=storage/
+    if [ ! -f "tests/$path" ]; then
+        path=${storage_prefix}${path}
+        if [ ! -f "tests/$path" ]; then
+            echo "No test $path found"
+            exit 1
+        fi
+    fi
+}
+
+vdsm_build()
+{
+    git clean -dxf && ./autogen.sh --system --enable-timestamp && \
```

Comment:
`git clean -dxf` is dangerous to do without proper warnings.  Maybe better to run something like  `make clean` after running autogen.sh (I do it in my builds)?

Reply:
git clean -dxf is dangerous, but it helps to teach user good practice by deleting files kept in the repo without adding them to git. Unfortunately without deep cleaning, sometimes the build is not correct.

The most annoying issue is when you have missing tag, and --enable-timestamp does not crate a build with a timestamp. You pull the missing tag but without cleaning the next build is wrong again.

This probably means that make clean is not strong enough, we may need to remove more files.

Reply:
I'd say is documented exactly like this in the READMEs. But I can change it if you know better options, or we can prompt the user for confirmation before executing.

Reply:
Having `git clean -dxf` in README is OK because the user must use it by hand and knows what's happening. It's different in a script. If any cleanup needed for a proper build is missing in `make clean` then it should be added there, rather than removing files created by the user (which can be anything -- notes, auxiliary scripts, testing data, ...).

Reply:
Removed the clean completely derived from the discussion in https://github.com/oVirt/vdsm/pull/242.